src/config/schema.ts

```diff
@@ -0,0 +1,337 @@
+      format: String,
+      default: '',
+      env: 'CUSTOM_CLOUDWATCH_LOG_GROUP',
+    },
+  },
+  core: {
+    port: {
+      doc: 'Application Port',
+      format: 'port',
+      default: defaults.app.port,
+      env: 'PORT',
+    },
+    otpLifeSpan: {
+      doc:
+        'OTP Life Span for Login. (Should be in miliseconds, e.g. 1000 * 60 * 15 = 15 mins)',
+      format: 'int',
+      default: defaults.login.otpLifeSpan,
+      env: 'OTP_LIFE_SPAN',
+    },
+    submissionsTopUp: {
+      doc: 'Number of submissions to top up submissions statistic by',
+      format: 'int',
+      default: 0,
+      env: 'SUBMISSIONS_TOP_UP',
+    },
+    nodeEnv: {
+      doc: 'Express environment mode',
+      format: [Environment.Prod, Environment.Dev, Environment.Test],
+      default: Environment.Prod,
+      env: 'NODE_ENV',
+    },
+  },
+}
+
+export const prodOnlyVarsSchema: Schema<IProdOnlyVarsSchema> = {
+  port: {
+    doc: 'SMTP port number',
+    format: 'port',
+    default: null,
+    env: 'SES_PORT',
+  },
+  host: {
+    doc: 'SMTP hostname',
+    format: String,
+    default: null,
+    env: 'SES_HOST',
+  },
+  user: {
+    doc: 'SMTP username',
+    format: String,
+    default: null,
+    env: 'SES_USER',
+  },
+  pass: {
+    doc: 'SMTP password',
+    format: String,
+    default: null,
+    env: 'SES_PASS',
+  },
+  dbHost: {
+    doc: 'Database URI',
+    format: (val) => {
+      // Will throw error if scheme and hosts are not present
+      const uriObject = mongodbUri.parse(val)
+      /*
+        e.g. mongodb://database:27017/formsg will be parsed into:
+        {
+          scheme: 'mongodb',
+          database: 'formsg',
+          hosts: [ { host: 'database', port: 27017 } ]
+        }
+        e.g. https://form.gov.sg will be parsed into: 
+        { 
+          scheme: 'https', 
+          hosts: [ { host: 'form.gov.sg' } ] 
+        }
+      */
+      if (uriObject.scheme !== 'mongodb') {
+        throw new Error('Scheme must be mongodb')
+      }
+      if (isNil(uriObject.database)) {
+        throw new Error('Database must be specified')
+      }
+    },
+    default: null,
+    env: 'DB_HOST',
+  },
+}
+
+export const loadS3BucketUrlSchema = (
+  isDev: boolean,
+): Schema<IBucketUrlSchema> => {
+  return {
+    attachmentBucketUrl: {
+      doc:
+        'Url of attachment S3 bucket derived from S3 endpoint and bucket name',
+      format: (val) =>
+        validateBucketUrl(val, { isDev, hasTrailingSlash: true }),
+      default: null,
+    },
+    logoBucketUrl: {
+      doc: 'Url of logo S3 bucket derived from S3 endpoint and bucket name',
+      format: (val) =>
+        validateBucketUrl(val, { isDev, hasTrailingSlash: false }),
+      default: null,
+    },
+    imageBucketUrl: {
+      doc: 'Url of images S3 bucket derived from S3 endpoint and bucket name',
+      format: (val) =>
+        validateBucketUrl(val, { isDev, hasTrailingSlash: false }),
+      default: null,
+    },
+  }
+}
```

Comment:
I noticed `CSP_REPORT_URI` isn't in here, is it supposed to be?

Reply:
yeap on purpose. Good catch. Moved it to the bonus feature ~ sentry